### DISCUSSION TITLE
Reed Solomon uncorrectable errors

### DISCUSSION THREAD

**Discussion by suterino** [mhostetter/galois#513] [General]
Hi all,
I am looking at the 
class galois.ReedSolomon

I am interested to analyze what happen when there are too many bad symbols beyond what is correctable,
for example: ReedSolomon(15, 9) can correct up to 3 symbols, what does it happen when there are 4 bad symbols?
most of the time it should be detectable, but not correctable.
I don't see how. 
I see 
galois.ReedSolomon.detect returns true or false
but it does not say how many bad symbols have been detected.
Shouldn't be a way to know if the errors have been detected but not corrected?
thank you

**Comment by donludovico** [comment]
This is a rather old topic, but I just ran across it. In your example, if you have 4 or more errors, two things can happen (since the code uses bounded minimum distance decoding):
* you are still more than 3 further errors away from any other (valid) codeword. Then you get a decoder failure, which you could detect in general (I have not taken a look at the source code yet, but I guess that this situation is currently not detected; this means you always end up in the next point)
* you are 3 or less further errors away from an other (valid) codeword. Then the decoder will correct the errors towards the closest (then wrong) codeword. You cannot detect this situation.
In general you must decide: you can either detect 6 errors (minimum Hamming distance d_min - 1) or correct floor((d_min-1)/2) errors (if you do not use Guruswami-Sudan decoding...). If you detect errors and then decide to correct them, you are trapped again, because the decoder (as a matter of fact) does not know which codeword was transmitted (if it new, there would be no need to decode, as it the already knew) and, hence, does not know onto wich codeword to correct to. The most sensible way is to decode onto the codeword that is the least number of errors away from the received one (as less errors are usually extremely more likely).
